--- xtables-addons-1.35/configure.ac.orig	2011-04-11 18:37:32.000000000 +0200
+++ xtables-addons-1.35/configure.ac	2011-05-06 20:03:27.286191084 +0200
@@ -40,44 +40,6 @@
 	-Wmissing-prototypes -Wredundant-decls -Wshadow -Wstrict-prototypes \
 	-Winline -pipe";
 
-if test -n "$kbuilddir"; then
-	AC_MSG_CHECKING([kernel version that we will build against])
-	krel="$(make -sC "$kbuilddir" M=. kernelrelease)";
-	kmajor="${krel%%[[^0-9]]*}";
-	kmajor="$(($kmajor+0))";
-	krel="${krel:${#kmajor}}";
-	krel="${krel#.}";
-	kminor="${krel%%[[^0-9]]*}";
-	kminor="$(($kminor+0))";
-	krel="${krel:${#kminor}}";
-	krel="${krel#.}";
-	kmicro="${krel%%[[^0-9]]*}";
-	kmicro="$(($kmicro+0))";
-	krel="${krel:${#kmicro}}";
-	krel="${krel#.}";
-	kstable="${krel%%[[^0-9]]*}";
-	kstable="$(($kstable+0))";
-	if test -z "$kmajor" -o -z "$kminor" -o -z "$kmicro"; then
-		echo "WARNING: Version detection did not succeed. Continue at own luck.";
-	else
-		echo "$kmajor.$kminor.$kmicro.$kstable in $kbuilddir";
-		if test "$kmajor" -gt 2 -o "$kminor" -gt 6 -o "$kmicro" -gt 38; then
-			echo "WARNING: You are trying a newer kernel. Results may vary. :-)";
-		elif test "$kmajor" -eq 2 -a "$kminor" -eq 6 -a "$kmicro" -ge 29; then
-			:; # everything ok
-		elif test \( "$kmajor" -lt 2 -o \
-		    \( "$kmajor" -eq 2 -a "$kminor" -lt 6 \) -o \
-		    \( "$kmajor" -eq 2 -a "$kminor" -eq 0 -a "$kmicro" -lt 17 \) -o \
-		    \( "$kmajor" -eq 2 -a "$kminor" -eq 6 -a "$kmicro" -eq 18 -a \
-		    "$kstable" -lt 5 \) \); then
-			echo "ERROR: That kernel version is not supported at all. Please see INSTALL for minimum configuration.";
-			exit 1;
-		else
-			echo "WARNING: That kernel version has been recently deprecated for use with Xtables-addons). Compilation may fail.";
-		fi;
-	fi;
-fi;
-
 AC_SUBST([regular_CPPFLAGS])
 AC_SUBST([regular_CFLAGS])
 AC_SUBST([kbuilddir])
