--- xtables-addons-3.3/extensions/pknock/xt_pknock.c~	2019-03-07 17:24:08.000000000 +0800
+++ xtables-addons-3.3/extensions/pknock/xt_pknock.c	2019-07-12 21:47:17.646586586 +0800
@@ -1125,7 +1125,9 @@
 
 	crypto.size = crypto_shash_digestsize(crypto.tfm);
 	crypto.desc.tfm = crypto.tfm;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 2, 0)
 	crypto.desc.flags = 0;
+#endif
 
 	pde = proc_mkdir("xt_pknock", init_net.proc_net);
 	if (pde == NULL) {
--- xtables-addons-3.3/extensions/xt_SYSRQ.c~	2019-03-07 17:24:08.000000000 +0800
+++ xtables-addons-3.3/extensions/xt_SYSRQ.c	2019-07-12 21:48:48.456584594 +0800
@@ -114,7 +114,9 @@
 	}
 
 	desc.tfm   = sysrq_tfm;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 2, 0)
 	desc.flags = 0;
+#endif
 	ret = crypto_shash_init(&desc);
 	if (ret != 0)
 		goto hash_fail;
