diff -rNpu xtables-addons-1.28-pre2.6.35-checkversion/extensions/compat_xtables.c xtables-addons-1.28/extensions/compat_xtables.c
--- xtables-addons-1.28-pre2.6.35-checkversion/extensions/compat_xtables.c	2010-07-24 11:33:11.000000000 +0200
+++ xtables-addons-1.28/extensions/compat_xtables.c	2010-09-01 09:23:44.000000000 +0200
@@ -105,7 +105,7 @@ static bool xtnu_match_check(const char 
 		return false;
 	if (nm->checkentry == NULL)
 		return true;
-	return nm->checkentry(&local_par);
+	return nm->checkentry(&local_par) == 0 ? true : false;
 }
 #endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 28) && \
@@ -313,7 +313,7 @@ static bool xtnu_target_check(const char
 	if (nt->checkentry == NULL)
 		/* this is valid, just like if there was no function */
 		return true;
-	return nt->checkentry(&local_par);
+	return nt->checkentry(&local_par) == 0 ? true : false;
 }
 #endif
 
