--- xtables-addons-1.10/extensions/GNUmakefile.in	2009-02-18 00:31:26.000000000 +0100
+++ xtables-addons-1.10.new/extensions/GNUmakefile.in	2009-02-18 20:31:45.005459510 +0100
@@ -63,7 +63,7 @@
 
 .PHONY: all install clean distclean FORCE
 
-all: subdirs modules user matches.man targets.man
+all: subdirs user matches.man targets.man
 
 subdirs:
 	@for i in ${subdirs_list}; do ${MAKE} -C $$i; done;
@@ -73,7 +73,7 @@
 
 user: ${targets}
 
-install: modules_install subdirs-install ${targets_install}
+install: subdirs-install ${targets_install}
 	@mkdir -p "${DESTDIR}${xtlibdir}";
 	install -pm0755 ${targets_install} "${DESTDIR}${xtlibdir}/";
 
