--- xtables-addons-1.5.7/extensions/GNUmakefile.in.orig	2008-09-01 21:35:51.000000000 +0200
+++ xtables-addons-1.5.7/extensions/GNUmakefile.in	2008-09-12 10:34:09.000000000 +0200
@@ -60,11 +60,11 @@
 
 .PHONY: all install clean distclean FORCE
 
-all: modules user matches.man targets.man
+all: user matches.man targets.man
 
 user: ${targets}
 
-install: modules_install ${targets_install}
+install: ${targets_install}
 	@mkdir -p "${DESTDIR}${xtlibdir}";
 	install -pm0755 ${targets_install} "${DESTDIR}${xtlibdir}/";
 
