--- xtables-addons-1.10/extensions/libxt_geoip.c	2009-02-18 00:31:26.000000000 +0100
+++ xtables-addons-1.10.new/extensions/libxt_geoip.c	2009-02-18 20:33:55.112397699 +0100
@@ -24,7 +24,7 @@
 #include <unistd.h>
 #include <xtables.h>
 #include "xt_geoip.h"
-#define GEOIP_DB_DIR "/var/geoip"
+#define GEOIP_DB_DIR "/usr/share/xt_geoip"
 
 static void geoip_help(void)
 {
--- xtables-addons-1.10/extensions/libxt_geoip.man	2009-02-18 00:31:26.000000000 +0100
+++ xtables-addons-1.10.new/extensions/libxt_geoip.man	2009-02-18 20:34:18.256330715 +0100
@@ -11,6 +11,6 @@
 .P
 The extra files you will need is the binary database files. They are generated
 from a country-subnet database with the geoip_csv_iv0.pl tool, available at
-http://jengelh.hopto.org/files/geoip/ . The files MUST be moved to /var/geoip/
+http://jengelh.hopto.org/files/geoip/ . The files MUST be moved to /usr/share/xt_geoip/
 as the shared library is statically looking for this pathname (e.g.
-/var/geoip/LE/de.iv0).
+/usr/share/xt_geoip/LE/de.iv0).
