--- xtables-addons-1.38/extensions/ipset-6/Makefile.am~	2011-08-20 17:02:31.000000000 +0200
+++ xtables-addons-1.38/extensions/ipset-6/Makefile.am	2011-08-26 13:15:36.983190997 +0200
@@ -8,12 +8,12 @@
 lib_LTLIBRARIES = libipset.la
 libipset_la_SOURCES = libipset/data.c libipset/icmp.c libipset/icmpv6.c \
                       libipset/mnl.c libipset/parse.c libipset/print.c \
-                      libipset/session.c libipset/types.c
+                      libipset/session.c libipset/types.c src/errcode.c
 libipset_la_LIBADD  = ${libmnl_LIBS}
 libipset_la_LDFLAGS = -version-info 1:0:0
 
 sbin_PROGRAMS = ipset
-ipset_SOURCES = src/ipset.c src/errcode.c src/ui.c src/ipset_bitmap_ip.c \
+ipset_SOURCES = src/ipset.c src/ui.c src/ipset_bitmap_ip.c \
                 src/ipset_bitmap_ipmac.c src/ipset_bitmap_port.c \
                 src/ipset_hash_ip.c src/ipset_hash_ipport.c \
                 src/ipset_hash_ipportip.c src/ipset_hash_ipportnet.c \
